bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,839 @@
+                scheduleDailyDataPoll();
+            }, delayToNextExecution, DAY_IN_MILLIS, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private String millisToLocalDateTime(final long milliseconds) {
+        LocalDateTime cvDate = Instant.ofEpochMilli(milliseconds).atZone(timeZoneProvider.getTimeZone())
+                .toLocalDateTime();
+        return cvDate.format(DateTimeFormatter.ofPattern(EXPECTED_TS_FORMAT));
+    }
+
+    private void cancelScheduleDailyDataPoll(final boolean allowInterrupt) {
+        synchronized (checkDailySchedulerLock) {
+            ScheduledFuture<?> job = dailyScheduler;
+            if (job != null) {
+                job.cancel(allowInterrupt);
+                dailyScheduler = null;
+            }
+        }
+    }
+
+    /**
+     * This handles the calculation of the delay until the next hourly forecast data should be retrieved.
+     * This calculates using a multiple since midnight to calculate when the next poll should occur.
+     * If the next poll is past midnight it is not scheduled as the daily synchronisation should reschedule a poll
+     * at that time.
+     */
+    private void scheduleNextHourlyForecastPoll() {
+        final long millisSinceDayStart = getMillisSinceDayStart();
+        final long pollRateMillis = config.hourlyForecastPollRate * 3600000L;
+        long initialDelayTimeToFirstCycle = pollRateMillis - (millisSinceDayStart % pollRateMillis);
+        initialDelayTimeToFirstCycle += RANDOM_GENERATOR.nextInt(60000);
+
+        if (initialDelayTimeToFirstCycle + millisSinceDayStart >= DAY_IN_MILLIS) {
+            logger.debug("Not scheduling poll after next daily cycle reset");
+        } else {
+            logger.debug("Can scheduling next Hourly forecast data poll to be in {} milliseconds at {}",
+                    initialDelayTimeToFirstCycle,
+                    millisToLocalDateTime(System.currentTimeMillis() + initialDelayTimeToFirstCycle));
+
+            // Schedule the first poll to occur after the given delay
+            hourlyForecastJob.scheduleExecution(initialDelayTimeToFirstCycle, () -> {
+                pollHourlyForecast();
+                scheduleNextHourlyForecastPoll();
+            });
+        }
+    }
+
+    /**
+     * This handles the calculation of the delay until the next hourly forecast data should be retrieved.
+     * This calculates using a multiple since midnight to calculate when the next poll should occur.
+     * If the next poll is past midnight it is not scheduled as the daily synchronisation should reschedule a poll
+     * at that time.
+     */
+    private void scheduleNextDailyForecastPoll() {
+        final long millisSinceDayStart = getMillisSinceDayStart();
+        long pollRateMillis = config.dailyForecastPollRate * 3600000L;
+        long initialDelayTimeToFirstCycle = pollRateMillis - (millisSinceDayStart % pollRateMillis);
```

Comment:
This very similar code is on atleast 5 places, i woudl suggest to add a convenience method:
```
    public static long timeTillNextStart(int pollRateInHours) {
        final long millisSinceDayStart = getMillisSinceDayStart();
        long pollRateMillis = pollRateInHours * 3600000L;
        return pollRateMillis - (millisSinceDayStart % pollRateMillis);
    }
```

That would make this and the other 5 calling places a one liner:
```suggestion
        long initialDelayTimeToFirstCycle = timeTillNextStart(config.dailyForecastPollRate);
```

Reply:
As part of the refactor this functionality was normalized. So hopefully this is achieved now 🤞